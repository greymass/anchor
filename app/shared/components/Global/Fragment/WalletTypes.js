// @flow
import React, { PureComponent } from 'react';
import { Button, Grid, Header } from 'semantic-ui-react';
import { translate } from 'react-i18next';

export class GlobalFragmentWalletTypes extends PureComponent<Props> {
  render() {
    const {
      onChange,
      t
    } = this.props;
    return (
      <Grid divided="vertically" padded="vertically" stackable>
        <Grid.Row columns={2}>
          <Grid.Column>
            <Header
              content="Hot Wallet"
              icon="lock"
              size="large"
              subheader="An account setup to perform actions on the blockchain using a locally encrypted private key. The security of a hot wallet is equal to the security of the computer."
            />
          </Grid.Column>
          <Grid.Column verticalAlign="middle" textAlign="center">
            <p>Import a private key to create a Hot Wallet.</p>
            <Button
              content="Import Private Key"
              icon="lock"
              onClick={onChange}
              primary
              size="large"
              value={1}
            />
          </Grid.Column>
        </Grid.Row>
        <Grid.Row columns={2}>
          <Grid.Column>
            <Header
              content="Hardware Wallet"
              icon="usb"
              size="large"
              subheader="An account that has been configured to use a hardware signing device. All keys are stored on the device and not on this computer."
            />
          </Grid.Column>
          <Grid.Column verticalAlign="middle" textAlign="center">
            <p>Load the public key(s) from a device to create a Ledger Wallet.</p>
            <Button
              content="Load Public Keys"
              icon="usb"
              onClick={onChange}
              primary
              size="large"
              value={2}
            />
          </Grid.Column>
        </Grid.Row>
        <Grid.Row columns={2}>
          <Grid.Column>
            <Header
              content="Watch Wallet"
              icon="eye"
              size="large"
              subheader="An account that cannot interact with the blockchain, but can monitor the account and create unsigned transactions for a cold wallet."
            />
          </Grid.Column>
          <Grid.Column verticalAlign="middle" textAlign="center">
            <p>Specify any account name to create a new watch wallet.</p>
            <Button
              content="Create Watch Wallet"
              icon="eye"
              onClick={onChange}
              primary
              size="large"
              value={3}
            />
          </Grid.Column>
        </Grid.Row>
      </Grid>
    );
  }
}

export default translate('global')(GlobalFragmentWalletTypes);
