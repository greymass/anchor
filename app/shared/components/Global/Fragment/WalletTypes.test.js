import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { GlobalFragmentWalletTypes } from './WalletTypes';

const mockProps = {
  onChange: jest.fn,
  t: jest.fn,
};

describe('GlobalFragmentWalletTypes', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<GlobalFragmentWalletTypes {...mockProps} />);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
