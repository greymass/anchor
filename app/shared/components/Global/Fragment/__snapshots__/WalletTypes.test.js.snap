// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GlobalFragmentWalletTypes renders correctly 1`] = `
<Grid
  divided="vertically"
  padded="vertically"
  stackable={true}
>
  <GridRow
    columns={2}
  >
    <GridColumn>
      <Header
        content="Hot Wallet"
        icon="lock"
        size="large"
        subheader="An account setup to perform actions on the blockchain using a locally encrypted private key. The security of a hot wallet is equal to the security of the computer."
      />
    </GridColumn>
    <GridColumn
      textAlign="center"
      verticalAlign="middle"
    >
      <p>
        Import a private key to create a Hot Wallet.
      </p>
      <Button
        as="button"
        content="Import Private Key"
        icon="lock"
        onClick={[Function]}
        primary={true}
        size="large"
        value={1}
      />
    </GridColumn>
  </GridRow>
  <GridRow
    columns={2}
  >
    <GridColumn>
      <Header
        content="Hardware Wallet"
        icon="usb"
        size="large"
        subheader="An account that has been configured to use a hardware signing device. All keys are stored on the device and not on this computer."
      />
    </GridColumn>
    <GridColumn
      textAlign="center"
      verticalAlign="middle"
    >
      <p>
        Load the public key(s) from a device to create a Ledger Wallet.
      </p>
      <Button
        as="button"
        content="Load Public Keys"
        icon="usb"
        onClick={[Function]}
        primary={true}
        size="large"
        value={2}
      />
    </GridColumn>
  </GridRow>
  <GridRow
    columns={2}
  >
    <GridColumn>
      <Header
        content="Watch Wallet"
        icon="eye"
        size="large"
        subheader="An account that cannot interact with the blockchain, but can monitor the account and create unsigned transactions for a cold wallet."
      />
    </GridColumn>
    <GridColumn
      textAlign="center"
      verticalAlign="middle"
    >
      <p>
        Specify any account name to create a new watch wallet.
      </p>
      <Button
        as="button"
        content="Create Watch Wallet"
        icon="eye"
        onClick={[Function]}
        primary={true}
        size="large"
        value={3}
      />
    </GridColumn>
  </GridRow>
</Grid>
`;
